lib/model/folder_sendrecv.go

```diff
@@ -315,18 +315,23 @@ func (f *sendReceiveFolder) processNeeded(dbUpdateChan chan<- dbUpdateJob, copyC
 			return true
 		}
 
+		if runtime.GOOS == "windows" && fs.WindowsInvalidFilename(intf.FileName()) {
+			f.newPullError(intf.FileName(), fs.ErrInvalidFilename)
+			return true
+		}
+
 		file := intf.(protocol.FileInfo)
 
+		// Do all checks that should not result in the pull being retried
+		// above this point.
+		changed++
```

Comment:
Previously we did not increment changed when a file had no availability. It seems to me this change might result in eternal retries of unavailable files (because devices are offline, for example).